scipy/optimize/tests/test__dual_annealing.py

```diff
@@ -336,3 +337,19 @@ def test_accept_reject_probabilistic(
         rate = 0 if pqv <= 0 else np.exp(np.log(pqv) / (1 - accept_param))
 
         assert_allclose(rate, accept_rate)
+
+    def test_bounds_class(self):
```

Comment:
One last thing that I just noticed - this test would be a lot stronger if the `bounds` were actually active at the optimal solution (i.e. at least one element of the solution should be at either an upper or lower bound). Otherwise, it's not clear (from the test alone) that the bounds actually affected the solution of the problem, and therefore we have little evidence (from the test alone) that the change is correct. Does that make sense?

Reply:
Yes, definitely makes sense. Adressed in last commit.

Reply:
Almost. Here's what I mean (below).